libs/react-helpers/src/lib/generic-data-provider.ts

```diff
@@ -13,9 +13,30 @@ export interface UpdateCallback<Data, Delta> {
     data: Data | null;
     error?: Error;
     loading: boolean;
+    pageInfo: PageInfo | null;
     delta?: Delta;
+    insertData?: Data | null;
+    totalCount?: number;
   }): void;
 }
+
+export interface Load<Data> {
+  (pagination: Pagination): Promise<Data | null>;
+}
+
+export interface Pagination {
+  first?: number;
+  after?: string;
+  last?: number;
+  before?: string;
+}
+
+export interface PageInfo {
+  startCursor?: string;
+  endCursor?: string;
+  hasNextPage?: boolean;
+  hasPreviousPage?: boolean;
+}
```

Comment:
since each prop is optional maybe would be little bit better like:

```
export interface Partial<Pagination> {
  first: number;
  after: string;
  last: number;
  before: string;
}

export interface Partial<PageInfo> {
  startCursor: string;
  endCursor: string;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
```